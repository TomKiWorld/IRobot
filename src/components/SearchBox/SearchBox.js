import React from 'react';

const SearchBox = ({ searchChange }) => {
  return (
    <div className='pa2'>
      <input
        aria-label='search-robots'
        className='pa3 ba b--green bg-lightest-blue'
        type='search'
        placeholder='search robots'
        onChange={(event) => searchChange(event.target.value)}
      />
    </div>
  );
}

export default SearchBox;
