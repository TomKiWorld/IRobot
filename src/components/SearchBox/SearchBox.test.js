import { shallow } from 'enzyme';
import React from 'react';
import SearchBox from './SearchBox';

describe('SearchBox Component', () => {
  const mockState = {
    search: ''
  }
  const changeSearch = (text) => {
    mockState.search = text;
  }
  const wrapper = shallow(<SearchBox searchChange={changeSearch}/>);
  it('Expect to render SearchBox Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  const searchInput = wrapper.find('[type=\'search\']').at(0);
  searchInput.simulate('change', { target: { value: 'Robot'} })
  it('Expect value to change', () => {
    expect(mockState.search).toEqual('Robot');
  });
});
