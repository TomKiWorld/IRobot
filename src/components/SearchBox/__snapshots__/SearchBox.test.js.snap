// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component Expect to render SearchBox Component 1`] = `ShallowWrapper {}`;
